test/test_unary_ufuncs.py

```diff
@@ -1,12 +1,151 @@
+               -.001, .001,
+               -.25, .25,
+               -1., 1.,
+               -math.pi / 2, math.pi / 2,
+               -math.pi + .00001, math.pi - .00001,
+               -math.pi, math.pi,
+               -math.pi - .00001, math.pi + .00001)
+_large_float_vals = (-501, 501,
+                     -1001.2, 1001.2,
+                     -13437.7, 13437.7,
+                     -4988429.2, 4988429.2,
+                     -1e20, 1e20)
+_float_extremals = (float('inf'), float('-inf'), float('nan'))
+
+
+def _make_tensor(size, device, dtype, *, low, high):
+    if dtype is torch.bool:
+        return torch.randint(0, 2, size, device=device, dtype=dtype)
+
+    def _maybe_clamp(t, low_default, low, high_default, high):
+        low = low_default if low is None else max(low_default, low)
+        high = high_default if high is None else min(high_default, high)
+        if low != low_default or high != high_default:
+            return torch.clamp(t, low, high)
+        return t
+
+    if dtype is torch.uint8:
+        t = torch.randint(0, 10, size, device=device, dtype=dtype)
+        return _maybe_clamp(t, 0, low, 10, high)
+    elif dtype in integral_types():
+        t = torch.randint(-9, 10, size, device=device, dtype=dtype)
+        return _maybe_clamp(t, -9, low, 10, high)
+    elif dtype in floating_types_and(torch.half, torch.bfloat16):
+        # Windows doesn't support torch.rand(bfloat16) on CUDA
+        if IS_WINDOWS and torch.device(device).type == 'cuda' and dtype is torch.bfloat16:
+            t = (torch.rand(size, device=device, dtype=torch.float32) * 18 - 9).to(torch.bfloat16)
+        else:
+            t = torch.rand(size, device=device, dtype=dtype) * 18 - 9
+        return _maybe_clamp(t, -9, low, 10, high)
+    else:
+        assert dtype in complex_types()
+        float_dtype = torch.float if dtype is torch.cfloat else torch.double
+        real = torch.rand(size, device=device, dtype=float_dtype) * 18 - 9
+        real = _maybe_clamp(real, -9, low, 10, high)
+        imag = torch.rand(size, device=device, dtype=float_dtype) * 18 - 9
+        imag = _maybe_clamp(imag, -9, low, 10, high)
+        return torch.complex(real, imag)
+
+
+# Returns an iterable of contiguous tensors with the same storage on the requested
+#   device and with the requested dtype. The iterable will include an empty
+#   tensor, scalar (0-dim) tensors, small 1D tensors, a medium 1D tensor, and
+#   a larger 2D tensor. The tensors will "cover" the values in vals, and if
+#   vals is None then dtype-specific values will be substituted.
+#   The tensors' elements, except for those specified in vals, will be in the
+#   domain [low, high), if specified.
+#   If include_large_values is False then larger values, like 500+, then the
+#     default values will not include larger values, like 500+.
+#   If include_extremal_values is False, then the default values will not include
+#     NaN, positive infinity, or negative infinity.
```

Comment:
It would be helpful for the comments here to explain how this function is intended to be used, as that would help inform what future extensions to the set under test are ok, and which are not.

Reply:
Updated.